impl/src/main/java/org/jboss/weld/util/Proxies.java

```diff
@@ -56,32 +58,37 @@
 
         private final List<Class<?>> interfaces;
         private final List<Class<?>> classes;
+        private final Map<Class<?>, String> classToPackageMap;
```

Comment:
BTW you're aware of the fact that `Class` does not implement equals()/hashCode() so the object identity will be used instead, right? ;-)

Reply:
Good point, that slipped me. I can actually easily make it `Map<String, String>` as further down the line we use the string representation anyway.